Extension/src/LanguageServer/configurations.ts

```diff
@@ -2001,28 +1998,39 @@ export class CppProperties {
                 continue;
             }
 
-            let resolvedPath: string = this.resolvePath(curPath);
-            if (!resolvedPath) {
-                continue;
-            }
+            // Cache the original value of each path to include any variables or path delimiting.
+            const originalPath: string = curPath;
+            const expandedPaths: string[] = this.resolveAndSplit([curPath], undefined, this.ExtendedEnvironment, true, false);
             let pathExists: boolean = true;
-            if (this.rootUri) {
-                const checkPathExists: any = util.checkPathExistsSync(resolvedPath, this.rootUri.fsPath + path.sep, isWindows, false);
-                pathExists = checkPathExists.pathExists;
-                resolvedPath = checkPathExists.path;
+            const incorrectExpandedPaths: string[] = [];
+
+            if (expandedPaths.length > 0) {
+                if (this.rootUri) {
+                    for (let expandedPath of expandedPaths) {
+                        expandedPath = this.resolvePath(expandedPath);
+                        const checkPathExists: any = util.checkPathExistsSync(expandedPath, this.rootUri.fsPath + path.sep, isWindows, false);
+                        if (!checkPathExists.pathExists) {
+                            // If there are multiple paths, store any non-existing paths to squiggle later on.
+                            incorrectExpandedPaths.push(expandedPath);
+                        }
+                    }
+                }
+            } else {
+                continue;
             }
+
             // Normalize path separators.
-            if (path.sep === "/") {
-                resolvedPath = resolvedPath.replace(/\\/g, path.sep);
+            if (path.sep === "/" && expandedPaths.length > 0) {
+                expandedPaths[0] = expandedPaths[0].replace(/\\/g, path.sep);
```

Comment:
Why only do this for the first one?  I feel like this should be in the loop for `expandedPaths`

Reply:
You're right, I was still thinking about the old implementation.